pkgs/applications/networking/avalanchego/default.nix

```
@@ -0,0 +1,52 @@
+{ buildGoModule
+, fetchFromGitHub
+, fetchpatch
+, lib
+, stdenv
+, IOKit
+}:
+
+buildGoModule rec {
+  pname = "avalanchego";
+  version = "1.9.5";
+
+  src = fetchFromGitHub {
+    owner = "ava-labs";
+    repo = pname;
+    rev = "v${version}";
+    hash = "sha256-Uim2Q3LvNEV0C5/gFdAiXKy2oKbhhc/ifRyz5BIh6zE=";
+  };
+buildInputs = lib.optionals stdenv.isDarwin [ IOKit ];
+
+  # patch zondax/hid dependency manually on darwin
+  patches = [(fetchpatch {
+    url = "https://github.com/ava-labs/avalanchego/pull/1194.patch";
+    name = "update-zondax-hid.patch";
+    hash = "";
+  })];
+
+  # go mod vendor has a bug, see: https://github.com/golang/go/issues/57529
+  proxyVendor = true;
+
+  subPackages = [ "main" ];
+
+  vendorHash = "";
```

```suggestion
vendorHash =
    if stdenv.isDarwin
    then "sha256-B3QC5IVVqyN59phG51dVm4glGZzLYt1mmlAxZXNsqTE="
    else "sha256-43bUvEwuppizLybj32eolSdpnqGJkkdA53/uBZkCrDc=";
```
Two vendor hashes. One for Linux without the patch, one for Darwin with the patch.

---

do we really end up generating different `vendorHash`es for different platforms? I would expect that we want to see the same go source for both

---

If we conditionally apply the patch, then we get two different vendor hashes because the dependencies are different, once with `github.com/zondax/hid v0.9.1-0.20220302062450-5552068d2266` (Linux, without the patch) and once with `github.com/zondax/hid v0.9.1` (Darwin with the patch).

---

I would have expected we would want to patch all platforms, not just darwin, especially since after it merges and is consumed, we will definitely see it on all platforms; we can just patch darwin, but it seems extra brittle?

---

Usually, I only patch the minimal needed changes. If a change is only needed for one platform, I guard it behind that platform because then I know for sure that it will still work on all other platforms. In addition, all future breaking changes with the patch will only break one platform.

But this is just my personal approach and I am fine with your solution.